src/fileio.cpp

```diff
@@ -1095,18 +1079,49 @@ void DetermineBasePaths(const char *exe)
 		homedir = (pw == nullptr) ? nullptr : stredup(pw->pw_dir);
 	}
 #endif
+	return homedir;
+}
+
+/**
+ * Determine the base (personal dir and game data dir) paths
+ * @param exe the path to the executable
+ */
+void DetermineBasePaths(const char *exe)
+{
+	char tmp[MAX_PATH];
+
+	char *homedir = GetHomeDir();
+
+#ifdef USE_XDG
+	const char *xdg_data_home = getenv("XDG_DATA_HOME");
+	if (xdg_data_home != nullptr) {
+		seprintf(tmp, lastof(tmp), "%s" PATHSEP "%s", xdg_data_home,
+				&PERSONAL_DIR[(PERSONAL_DIR[0] == '.')]);
+	} else {
+		/* Defaults to ~/.local/share */
+		seprintf(tmp, lastof(tmp), "%s" PATHSEP ".local" PATHSEP "share" PATHSEP "%s", homedir,
```

Comment:
This assumes "homedir != nullptr".
Maybe a std::string is indeed easier :)

Reply:
This is now checked it seems, but now if `homedir == nullptr`, `tmp` will be left unitialized. In the old code, I think a similar thing happened, though, when `xdgDataHome` would return `nullptr` (assuming it could).